### PR TITLE
Detect dark theme via css variable

### PR DESCRIPTION
**Pull request by Theta-Dev**
I have just created a custom dark theme for my Gitea instance. It looks good, but the Monaco text editor did not pick up the dark theme like it does with the `arc-green` theme enabled.

So I looked into the source and found out that the dark editor theme only gets enabled when the builtin `arc-green` or `auto` theme is chosen. Any additional themes are treated as light themes.

That's why I updated the `isDarkTheme()` function to look for the `--is-dark-theme` css variable. If that variable is set to `true`, the editor dark mode gets enabled.

And because CSS variables can be set within queries, too, responsive themes like `auto` are possible without additional js logic.

### CODE REVIEW

**Comment by silverwind** [comment]
Seems like a good approach, the old way with the name check was a hack. I'm not sure about the removal of the `theme-auto` condition thought, maybe it should be kept.

File: web_src/js/utils.js [hunk 1]
```diff
@@ -26,13 +26,9 @@ export function isObject(obj) {
...
 // returns whether a dark theme is enabled
 export function isDarkTheme() {
-  if (document.documentElement.classList.contains('theme-auto')) {
-    return window.matchMedia('(prefers-color-scheme: dark)').matches;
-  }
-  if (document.documentElement.classList.contains('theme-arc-green')) {
-    return true;
-  }
-  return false;
+  const isDarkTheme = window.getComputedStyle(document.documentElement)
```
Comment (Gusted) [inline web_src/js/utils.js]: I think this is a nice addition - as custom CSS themes which are made for gitea can now also specify that their theme is a dark theme via this way. 

But now we're removing the `theme-auto`, so a new media query in the CSS could be made to handle that part.
```css
@media (prefers-color-sheme: dark) {
 :root {
  --is-dark-theme: true;
 }
}
```

Comment (Gusted) [reply web_src/js/utils.js]: By @silverwind's comment, this could be a nvm.

Comment (Theta-Dev) [reply web_src/js/utils.js]: Yes, the `auto` theme loads the entire `arc-green` theme including the `--is-dark-theme` variable if dark mode is enabled.

https://github.com/go-gitea/gitea/blob/main/web_src/less/themes/theme-auto.less

I have tested it and it works as expected.

**Review by Gusted — APPROVED** [review]

File: web_src/js/utils.js [hunk 2]
```diff
@@ -26,13 +26,9 @@ export function isObject(obj) {
...
+    .getPropertyValue('--is-dark-theme').trim().toLowerCase();
+  return isDarkTheme === 'true';
```
Comment (silverwind) [inline web_src/js/utils.js]:
```suggestion
  const stype = window.getComputedStyle(document.documentElement);
  return style.getPropertyValue('--is-dark-theme').trim().toLowerCase() === 'true';
```

minor refactor

Comment (Theta-Dev) [reply web_src/js/utils.js]: okay. done it.

**Review by silverwind — APPROVED** [review]

File: web_src/js/utils.js [hunk 3]
```diff
@@ -26,13 +26,8 @@ export function isObject(obj) {
...
 export function isDarkTheme() {
-  if (document.documentElement.classList.contains('theme-auto')) {
-    return window.matchMedia('(prefers-color-scheme: dark)').matches;
-  }
-  if (document.documentElement.classList.contains('theme-arc-green')) {
-    return true;
-  }
-  return false;
+  const style = window.getComputedStyle(document.documentElement);
+  return style.getPropertyValue('--is-dark-theme').trim().toLowerCase() === 'true';
```
Comment (delvh) [inline web_src/js/utils.js:30]: What is the behavior of `getPropertyValue` when the variable is not set?
The empty string?

Comment (Gusted) [reply web_src/js/utils.js:30]: It will return a empty string.

Comment (delvh) [reply web_src/js/utils.js:30]: Then ignore this comment.
I would resolve it, but for whatever reason, GitHub does not allow to resolve your own comments.

Comment (Theta-Dev) [reply web_src/js/utils.js:30]: Yes, it is.

```js
window.getComputedStyle(document.documentElement).getPropertyValue('--is-dark-theme')
''
```

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**